backend/app/adapter/gqlapi/resolver/authmutation.go

```diff
@@ -77,6 +97,45 @@ func (a AuthMutation) CreateURL(args *CreateURLArgs) (*URL, error) {
 	}
 }
 
+// UpdateURLArgs represents the possible parameters for updateURL endpoint
+type UpdateURLArgs struct {
+	OldAlias string
+	URL      URLInput
+}
+
+func (u *URLInput) createUpdate() *entity.ShortLink {
+	if u.isEmpty() {
+		return nil
+	}
+
+	return &entity.ShortLink{
+		Alias:    u.customAlias(),
+		LongLink: u.originalURL(),
+		ExpireAt: u.ExpireAt,
+	}
+
+}
+
+// UpdateURL updates a short link mapping that belongs to a user
+func (a AuthMutation) UpdateURL(args *UpdateURLArgs) (*URL, error) {
+	user, err := viewer(a.authToken, a.authenticator)
+	if err != nil {
+		return nil, ErrInvalidAuthToken{}
+	}
+
+	update := args.URL.createUpdate()
+	if update == nil {
+		return nil, err
```

Comment:
It seems a bit weird to return nil for both of them maybe the error would be better in this scenario so that someone reading this line later is not confused by it. I had previously created an empty update error for this, would the error be worded better as something else?

Reply:
@oatovar Either makes sense. Should empty update be an error?

Reply:
For right now just returning nil will suffice I believe :sweat_smile: